project/JdkOptions.scala

```diff
@@ -74,9 +68,7 @@ object JdkOptions extends AutoPlugin {
       jdk8home,
       fullJavaHomes,
       Nil,
-      // '-release 8' would be a neater option here, but is currently not an
-      // option because it doesn't provide access to `sun.misc.Unsafe` #27079
-      (java8home: File) => Seq("-source", "8", "-target", "8", "-bootclasspath", java8home + "/jre/lib/rt.jar"))
+      (java8home: File) => Seq("-source", "8", "-target", "8"))
```

Comment:
Unfortunately, I don't think this is equivalent: if a signature would have changed between jdk8 and jdk11, `-target 8` would emit jdk8-compatible bytecode, but would still use the jdk11 signature - so the jar would not work on an actual jdk8.

On the one hand I don't find any instances of that in our current codebase, but on the other hand I'm not sure it's worth introducing the risk...

Reply:
Thanks for the pickup, indeed this problem seems like a different (but slightly unrelated one). I think for now I will revert the change and make an issue for it to check later in order to verify if its needed.

Reply:
It will not use the **JDK 11** signature because it was compiled targeting the **JDK 8**. It can't even see such a signature.
That is the whole point of the `release` flag, to ensure that both the bytecode and linked **Java** lib match the target **JDK** version.
See https://github.com/scala/scala/pull/9982

Reply:
So also according to https://github.com/scala/bug/issues/12643 which describes the same problem,  this workaround shouldn't be necessary.

It appears to be quite easy to bisect this to confirm this, will look tonight

Reply:
> That is the whole point of the `release` flag, to ensure that both the bytecode and linked Java lib match the target JDK version.
See https://github.com/scala/scala/pull/9982

You seem to be referring to the scalac flags, but this line is about javac flags.

Also on javac, indeed that would be the point of the `--release` flag. However here we use the `-target` flag, which only affects the bytecode and not the signatures on the classpath.

The "correct" fix would be to use the `--release` flag, but that is not yet (easily) possible as that does not allow access to `sun.misc.Unsafe`, so it wouldn't be able to compile `actor/src/main/java/org/apache/pekko/util/Unsafe.java`. AFAICS that did not change with https://github.com/scala/bug/issues/12643.

Reply:
I was checked the bytecode in https://github.com/akka/akka/pull/31652 ，what's the result with this pr？

Reply:
> @som-snytt Since you made the PR at https://github.com/scala/scala/pull/9982 maybe you can comment on this?

you should use the -release option

Reply:
> I was checked the bytecode in [akka/akka#31652](https://github.com/akka/akka/pull/31652) ，what's the result with this pr？

I think this is a separate issue, I already checked the bytecode in https://github.com/apache/incubator-pekko/pull/258#issuecomment-1477908162

Reply:
> @som-snytt Since you made the PR at https://github.com/scala/scala/pull/9982 maybe you can comment on this?

https://github.com/scala/scala/pull/10109

Reply:
I'm not sure which issue I'm pinged about, but I remember 2.12 targets 8, making Unsafe available is a pain, and also if there is a subtle breakage in Java API then you may want to compile against rt.jar instead of using `--release` (which uses the `ct.sym` file).

Reply:
> I'm not sure which issue I'm pinged about, but I remember 2.12 targets 8, making Unsafe available is a pain, and also if there is a subtle breakage in Java API then you may want to compile against rt.jar instead of using `--release` (which uses the `ct.sym` file).

So it is specifically about targeting `rt.jar` with javac `-target` command because of unsafe, i.e. `(java8home: File) => Seq("-source", "8", "-target", "8", "-bootclasspath", java8home + "/jre/lib/rt.jar"))`.

It does sound like from what you are saying that we should just leave this.

Reply:
Yes also I confirmed the deprecation can be silenced by `-Wconf`. I think I just saw on that scala ticket that I said maybe undeprecate target if it has a mainstream use case still, but no action was taken. I think `-Wconf` is not terrible.

Reply:
Someone had a different need to compile against rt.jar, and https://github.com/scala/scala/pull/10336 has guidance if you're building on later jdk but against old rt.jar (jdk 8 by definition).

Reply:
> Someone had a different need to compile against rt.jar, and [scala/scala#10336](https://github.com/scala/scala/pull/10336) has guidance if you're building on later jdk but against old rt.jar (jdk 8 by definition).

Perfect thanks, I will just make an issue and reference that PR in it.

Reply:
I will mark this as resolved, made an issue at https://github.com/apache/incubator-pekko/issues/262 where we can investigate the removal of this workaround separately. Most critical thing for this PR specifically is to update scalac versions because of a CVC.

@raboof @BalmungSan @He-Pin @som-snytt Thanks a ton for the help/clarification